browser/ui/views/crash_report_permission_ask_dialog_view.cc

```diff
@@ -0,0 +1,261 @@
+      footnote, ui::NativeTheme::kColorId_DialogBackground));
+
+  const std::u16string setting_text = l10n_util::GetStringUTF16(
+      IDS_CRASH_REPORT_PERMISSION_ASK_DIALOG_FOOTNOTE_TEXT_SETTING_PART);
+  const std::u16string footnote_text = l10n_util::GetStringFUTF16(
+      IDS_CRASH_REPORT_PERMISSION_ASK_DIALOG_FOOTNOTE_TEXT, setting_text,
+      &offset);
+  auto* footnote_label =
+      footnote->AddChildView(std::make_unique<views::StyledLabel>());
+  footnote_label->SetHorizontalAlignment(gfx::ALIGN_CENTER);
+  footnote_label->SetText(footnote_text);
+
+  views::StyledLabel::RangeStyleInfo setting_style =
+      views::StyledLabel::RangeStyleInfo::CreateForLink(
+          base::BindRepeating(OpenSettingPage));
+  footnote_label->AddStyleRange(
+      gfx::Range(offset, offset + setting_text.length()), setting_style);
+
+  views::StyledLabel::RangeStyleInfo footnote_default_style;
+  constexpr int kFootnoteFontSize = 12;
+  footnote_default_style.custom_font =
+      GetFont(kFootnoteFontSize, gfx::Font::Weight::NORMAL);
+
+  if (offset != 0)
+    footnote_label->AddStyleRange(gfx::Range(0, offset),
+                                  footnote_default_style);
+
+  if (offset + setting_text.length() != footnote_text.length())
+    footnote_label->AddStyleRange(
+        gfx::Range(offset + setting_text.length(), footnote_text.length()),
+        footnote_default_style);
+}
+
+std::unique_ptr<views::NonClientFrameView>
+CrashReportPermissionAskDialogView::CreateNonClientFrameView(
+    views::Widget* widget) {
+  if (!use_custom_frame())
+    return DialogDelegateView::CreateNonClientFrameView(widget);
+
+  views::LayoutProvider* provider = views::LayoutProvider::Get();
+  auto frame = std::make_unique<views::BubbleFrameView>(
+      provider->GetInsetsMetric(views::INSETS_DIALOG_TITLE), gfx::Insets());
+
+  const views::BubbleBorder::Shadow kShadow =
+      views::BubbleBorder::DIALOG_SHADOW;
+  std::unique_ptr<views::BubbleBorder> border =
+      std::make_unique<views::BubbleBorder>(views::BubbleBorder::FLOAT, kShadow,
+                                            gfx::kPlaceholderColor);
+  border->set_use_theme_background_color(true);
+  constexpr int kCornerRadius = 16;
+  border->SetCornerRadius(kCornerRadius);
+  frame->SetFootnoteView(DisownFootnoteView());
+  frame->SetBubbleBorder(std::move(border));
+  return frame;
+}
+
+ui::ModalType CrashReportPermissionAskDialogView::GetModalType() const {
+  return ui::MODAL_TYPE_WINDOW;
+}
+
+bool CrashReportPermissionAskDialogView::ShouldShowCloseButton() const {
+  return false;
+}
+
+bool CrashReportPermissionAskDialogView::ShouldShowWindowTitle() const {
+  return false;
+}
+
+void CrashReportPermissionAskDialogView::OnDialogInitialized() {
+  SetButtonRowInsets(gfx::Insets(0, 0, 18, 24));
+}
+
+void CrashReportPermissionAskDialogView::OnAcceptButtonClicked() {
+  // Enable crash reporting.
+  ChangeMetricsReportingState(true);
```

Comment:
@iefremov can you check this? to enable crash reporting, calling this is sufficient?